trin-core/src/portalnet/overlay_service.rs

```diff
@@ -560,7 +668,16 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
             .as_ref()
             .and_then(|m| Some(m.report_inbound_find_nodes()));
         let distances64: Vec<u64> = request.distances.iter().map(|x| (*x).into()).collect();
-        let enrs = self.nodes_by_distance(distances64);
+
+        // Currently limiting to 3 ENRs until multiple messages is supported.
+        let enrs = {
+            let enrs = self.nodes_by_distance(distances64);
+            if enrs.len() > 3 {
+                enrs[0..3].to_vec()
+            } else {
+                enrs
+            }
+        };
```

Comment:
While we don't yet support multiple `Nodes` responses, we should not limit the number of ENRs to return to 3. Based on [this spec issue](https://github.com/ethereum/portal-network-specs/issues/136), which will inform the longer term strategy, we should be able to return 8 or 9.

We can use the [`size` method of `Enr`](https://docs.rs/enr/0.5.1/enr/struct.Enr.html#method.size) to determine the maximum number of ENRs that we can fit in a single message. This is better than relying on a heuristic.